### PR TITLE
add nccl2 dist tests

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

**Review by gongweibao — DISMISSED** [review]
Need to add more nccl update test case.

File: python/paddle/fluid/tests/unittests/test_dist_base.py [hunk 1]
```diff
@@ -372,7 +396,71 @@ def _run_cluster(self, model, envs, check_error_log):
...
+        if self._mem_opt:
+            tr0_cmd += " --mem_opt"
+            tr1_cmd += " --mem_opt"
+        if self._use_reduce:
+            tr0_cmd += " --use_reduce"
+            tr1_cmd += " --use_reduce"
+        if self._use_reader_alloc:
+            tr0_cmd += " --use_reader_alloc"
+            tr1_cmd += " --use_reader_alloc"
+        if self.__use_cuda:
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/test_dist_base.py:423]: _use_cuda?

File: python/paddle/fluid/tests/unittests/test_dist_base.py [hunk 2]
```diff
@@ -372,7 +396,71 @@ def _run_cluster(self, model, envs, check_error_log):
...
+            tr0_cmd += " --use_cuda"
+            tr1_cmd += " --use_cuda"
+            env0 = {"CUDA_VISIBLE_DEVICES": "0"}
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/test_dist_base.py:426]: Do we need to add select_gpu envs?

Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/test_dist_base.py:426]: Not for this test case, will add multi proc test case in next PR

**Review by gongweibao — APPROVED** [review]
LGTM

**Pull request merged**